packages/block-library/src/site-title/edit/index.js

```diff
@@ -63,12 +66,18 @@ export default function SiteTitleEdit( {
 		</TagName>
 	) : (
 		<TagName { ...blockProps }>
-			<a
-				href="#site-title-pseudo-link"
-				onClick={ ( event ) => event.preventDefault() }
-			>
-				{ readOnlyTitle || __( 'Site Title placeholder' ) }
-			</a>
+			{ isLink ? (
+				<a
+					href="#site-title-pseudo-link"
+					onClick={ ( event ) => event.preventDefault() }
+				>
+					{ readOnlyTitle || __( 'Site Title placeholder' ) }
+				</a>
+			) : (
+				<div style={ { display: 'inline-block' } }>
```

Comment:
Why this `div` is needed? Can we handle this `inline-block` if needed with css instead or at the parent element?

>const TagName = level === 0 ? 'p' : `h${ level }`;

In the case of `level:0` and no link, will add a `div` inside a `p`, which is not valid, no?

Reply:
You are right, `div` is not valid inside a heading or `p`.